core/src/driver/scheduler/system.rs

```diff
@@ -192,8 +196,17 @@ fn log_submit_result(batch_id: BatchId, result: &Result<()>) {
     }
 }
 
+fn duration_until_healthy(now: SystemTime) -> Duration {
+    let next_start_solve_time = BatchId::current(now).unwrap().solve_start_time();
```

Comment:
I think it would be nice here if we return immediately in case we are before `target_start_solve_time` of `BatchId::currently_being_solved`. 
Or do you think the added complexity is not worth the benefit of saving 5 minutes of ready time in 1/6 of the cases (assuming uniform distribution of restarts)?

Reply:
Like you said I didn't think this was worth it and it's inline with the evm scheduler which doesn't use the target solve start time. It's also a little safer in the case we are very close to target_start_solve_time but the time between the two pods is out of sync. Then it could happen that the other pod started solving already even though we think it is safe for us to start.

Reply:
Could using `target_start_solve_time` become an issue if the configuration changes?

Reply:
It would have to use the value from the configuration. I'm not to worried about system time between pods being dramatically out of sync ( we can cancel an ongoing processing unless we are already sending the tx), but I can see how the added logic is not worth the benefit (we can see if long wait times are bothering us).